scripts/wrapper/withdraw.js

```diff
@@ -12,34 +13,62 @@ module.exports = function (web3, artifacts) {
   } = require("../utils/trading_strategy_helpers")(web3, artifacts)
 
   const assertGoodArguments = function (argv) {
+    if (!argv.masterSafe) throw new Error("Argument error: --masterSafe is required")
+
     if (!argv.requestWithdraw && !argv.withdraw && !argv.transferFundsToMaster) {
       throw new Error("Argument error: one of --requestWithdraw, --withdraw, --transferFundsToMaster must be given")
     } else if (argv.requestWithdraw && (argv.transferFundsToMaster || argv.withdraw)) {
       throw new Error("Argument error: --requestWithdraw cannot be used with any of --withdraw, --transferFundsToMaster")
     }
+
+    if (!argv.withdrawalFile && !argv.from) {
+      throw new Error("Argument error: one of --withdrawalFile, --from must be given")
+    } else if (argv.withdrawalFile && argv.from) {
+      throw new Error("Argument error: --from cannot be used with --withdrawalFile")
+    }
+
+    if (argv.from) {
+      if (!argv.tokens && !argv.tokenIds) {
+        throw new Error("Argument error: one of --tokens, --tokenIds must be given when using --from")
+      } else if (argv.tokens && argv.tokenIds) {
+        throw new Error("Argument error: only one of --tokens, --tokenIds is required when using --from")
+      }
+    } else {
+      if (argv.tokens || argv.tokenIds) {
+        throw new Error("Argument error: --tokens or --tokenIds can only be used with --from")
+      }
+    }
   }
 
-  const getMaxWithdrawableAmount = async function (argv, bracketAddress, tokenInfo, exchange, printOutput = false) {
+  const getMaxWithdrawableAmount = async function (
+    argv,
+    bracketAddress,
+    tokenData,
+    exchange,
+    currentBatchId,
+    printOutput = false
+  ) {
     const log = printOutput ? (...a) => console.log(...a) : () => {}
     let amount
-    const token = tokenInfo.instance
-    if (argv.requestWithdraw) amount = (await exchange.getBalance(bracketAddress, tokenInfo.address)).toString()
+    const token = tokenData.instance
+    if (argv.requestWithdraw) amount = (await exchange.getBalance(bracketAddress, tokenData.address)).toString()
     else if (argv.withdraw) {
-      const currentBatchId = Math.floor(Date.now() / (5 * 60 * 1000)) // definition of BatchID, it avoids making a web3 request for each withdrawal to get BatchID
```

Comment:
Although this is now gone, I wanted to point out that this appears to be the wrong calculation for batch ID. Shouldn't it be ` Math.floor(Date.now() / 300)` (i.e. without multiplication by 1000)